### PR TITLE
Create Dataset

### PR DESCRIPTION
**Pull request by falcowinkler**
I also added pandas and a notebook, i thought it could be helpful to play with the dataset a bit

plus i added a simple example of under defined autoencoder,
but i don't understand so well the result we want to achieve

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -9,6 +9,7 @@ numpy = "==1.14.5"
 s3fs = "*"
 monty = {editable = true, path = "."}
 pandas = "*"
+"e1839a8" = {path = ".", editable = true}
```
Comment (meandor) [inline Pipfile]: redundant with line 10

File: monty/data.py [hunk 2]
```diff
@@ -2,14 +2,14 @@
 import os
 
 import s3fs
+import tensorflow as tf
 
 logger = logging.getLogger(__name__)
 
 
-def download_if_not_present():
+def download_if_not_present(local_file_path='resources/PBMC.csv'):
```
Comment (meandor) [inline monty/data.py]: would be nicer to not hide parameters, so less default values in this case

File: monty/data.py [hunk 3]
```diff
@@ -18,3 +18,18 @@ def download_if_not_present():
 
     logger.info('done downloading dataset')
     return local_file_path
+
+
+def create_dataset(path_to_raw_dataset, num_epochs=10, batch_size=128, shuffle_buffer_size=1000, row_count=2700):
```
Comment (meandor) [inline monty/data.py]: * default params again
* path_to_raw_dataset -> csv_file

File: monty/data.py [hunk 4]
```diff
@@ -18,3 +18,18 @@ def download_if_not_present():
...
+    filenames = [path_to_raw_dataset]
+    record_defaults = [tf.int64] * row_count
+    dataset = tf.contrib.data.CsvDataset(filenames, record_defaults, header=True,
```
Comment (meandor) [inline monty/data.py]: inline filenames here

File: monty/data.py [hunk 5]
```diff
@@ -18,3 +18,18 @@ def download_if_not_present():
...
+                                         select_cols=[x for x in range(1, row_count + 1)])
```
Comment (meandor) [inline monty/data.py]: no list comprehension needed:
[x for x in range(1, row_count + 1)] = range(1, row_count + 1)

File: tests/test_data.py [hunk 6]
```diff
@@ -1,27 +1,40 @@
 import os
 
+import tensorflow as tf
 from monty import data
 
+test_file_path = 'resources/PBMC_test.csv'
```
Comment (meandor) [inline tests/test_data.py]: make it "constant" -> UPPER_CASE

File: tests/test_data.py [hunk 7]
```diff
@@ -1,27 +1,40 @@
...
-    assert data.download_if_not_present() == 'resources/PBMC.csv'
+    assert data.download_if_not_present(local_file_path=test_file_path) == test_file_path
     assert mocked_s3.call_count == 0
+
+
+def test_create_dataset():
+    iterator = data.create_dataset("test_resources/PBMC_test.csv", batch_size=4, shuffle_buffer_size=8)
+    sess = tf.Session()
+    with sess.as_default():
+        tf.train.start_queue_runners(sess)
```
Comment (meandor) [inline tests/test_data.py]: what do you need the queue here for?

File: tests/test_data.py [hunk 8]
```diff
@@ -1,27 +1,40 @@
...
+        first_batch = sess.run(iterator)
+        print(first_batch)
```
Comment (meandor) [inline tests/test_data.py]: no print here plox

**Review by meandor — CHANGES_REQUESTED** [review]
please take a look at the review

File: monty/data.py (hunk 5 above)
Comment (falcowinkler) [reply monty/data.py]: nope, range returns a range object not a list and tf wants a list :(

Comment (falcowinkler) [reply monty/data.py]: list(range(1, row_count + 1)) will do though

**Pull request merged**